File: PaintHouse.py
class Solution(object):
    def minCost(self, costs):
        """
        :type costs: List[List[int]]
        :rtype: int
        """
        r,b,g = 0,0,0
        for nhouse in costs:
            tmpr = min(b,g) + nhouse[0]
            tmpb = min(r,g) + nhouse[1]
            tmpg = min(r,b) + nhouse[2]
            r,b,g = tmpr,tmpb,tmpg
        return min(r,b,g)

File: paintHouse.kt
class Solution {
    fun minCost(costs: Array<IntArray>): Int {
        var rgb = IntArray(3)
        costs.forEach{cost ->
            val nextRed   = Math.min(rgb[1], rgb[2]) + cost[0]
            val nextBlue  = Math.min(rgb[0], rgb[2]) + cost[1]
            val nextGreen = Math.min(rgb[0], rgb[1]) + cost[2]
            rgb[0] = nextRed
            rgb[1] = nextBlue
            rgb[2] = nextGreen
        }
        return rgb.min()
    }
}
